core/src/test/java/org/apache/iceberg/catalog/CatalogTests.java

```diff
@@ -319,6 +402,1345 @@ public void testNamespaceWithDot() {
     Assert.assertFalse("Namespace should not exist", catalog.namespaceExists(withDot));
   }
 
+  @Test
+  public void testBasicCreateTable() {
+    C catalog = catalog();
+
+    TableIdentifier ident = TableIdentifier.of("ns", "table");
+
+    Assert.assertFalse("Table should not exist", catalog.tableExists(ident));
+
+    Table table = catalog.buildTable(ident, SCHEMA).create();
+    Assert.assertTrue("Table should exist", catalog.tableExists(ident));
+
+    // validate table settings
+    Assert.assertEquals("Table name should report its full name", catalog.name() + "." + ident, table.name());
+    Assert.assertEquals("Schema should match expected ID assignment",
+        TABLE_SCHEMA.asStruct(), table.schema().asStruct());
+    Assert.assertNotNull("Should have a location", table.location());
+    Assert.assertTrue("Should be unpartitioned", table.spec().isUnpartitioned());
+    Assert.assertTrue("Should be unsorted", table.sortOrder().isUnsorted());
+    Assert.assertNotNull("Should have table properties", table.properties());
+  }
+
+  @Test
+  public void testBasicCreateTableThatAlreadyExists() {
+    C catalog = catalog();
+
+    TableIdentifier ident = TableIdentifier.of("ns", "table");
+
+    Assert.assertFalse("Table should not exist", catalog.tableExists(ident));
+
+    catalog.buildTable(ident, SCHEMA).create();
+    Assert.assertTrue("Table should exist", catalog.tableExists(ident));
+
+    AssertHelpers.assertThrows("Should fail to create a table that already exists",
+        AlreadyExistsException.class, "ns.table",
+        () -> catalog.buildTable(ident, OTHER_SCHEMA).create());
+
+    Table table = catalog.loadTable(ident);
+    Assert.assertEquals("Schema should match original table schema",
+        TABLE_SCHEMA.asStruct(), table.schema().asStruct());
+  }
+
+  @Test
+  public void testCompleteCreateTable() {
+    C catalog = catalog();
+
+    TableIdentifier ident = TableIdentifier.of("ns", "table");
+
+    Assert.assertFalse("Table should not exist", catalog.tableExists(ident));
```

Comment:
I think the only thing that could be extracted out would be the create table step. However, some of these tests only create a namespace and not the table. Others create a table in a different namespace, using the added `TABLE` constant.

We could make a class level constant out of `ident`, but it might confuse things with `TABLE`.

Is it possible to use that instead? I'm assuming there's a reason for the change in namespace.